tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Steps/EmailSteps.cs

```diff
@@ -37,46 +44,116 @@ public async Task EmailIsNotSent()
             emails.Should().BeNullOrEmpty();
         }
 
-        [Then(@"the email sent contains the following information")]
-        public async Task ThenEmailContains(Table table)
+        [Then(@"the email contains the following information")]
+        public async Task ThenTheEmailContainsTheFollowingInformation(Table table)
         {
             var expectedContents = table.CreateSet<EmailContentsTable>().First();
 
             var expected = new
             {
-                Attachments = new List<EmailAttachmentData>
-                {
-                    new EmailAttachmentData(
-                        Encoding.UTF8.GetBytes($"{expectedContents.AttachmentHeader1}\r\n{expectedContents.OrderId}\r\n"),
-                        expectedContents.FileName,
-                        new ContentType("text/csv"))
-                },
-                From = new List<EmailAddress>
-                {
-                    new EmailAddress(Empty, expectedContents.From)
-                },
+                From = new List<EmailAddress> { new EmailAddress(Empty, expectedContents.From) },
+                To = new List<EmailAddress> { new EmailAddress(Empty, expectedContents.To) },
                 expectedContents.Text,
                 expectedContents.Subject,
-                To = new List<EmailAddress>
-                {
-                    new EmailAddress(Empty, expectedContents.To)
-                }
             };
-            
+
             var actual = (await _emailServerDriver.FindAllEmailsAsync()).First();
 
             actual.Should().BeEquivalentTo(expected, conf => conf.IncludingAllDeclaredProperties());
         }
 
+        [Then(@"the patient numbers price type attachment contains the correct information")]
+        public async Task ThenThePatientNumbersPriceTypeAttachmentContainsTheCorrectInformation()
+        {
```

Comment:
One suggestion here would be to refactor the code within the step to make it a bit more readable. It looks like there are two assertions here one for the header and one for the body of the attachment. To be honest if could use something like the "Request" / "Response" type construct we have used for some of the other steps e.g. `PurchaseDocumentEmailResponse` that could read the data out of the email & the step can then easily assert the information.

Also we will need some way of identifying the email attachment as won't be able to rely on the "first" when the next story is completed.

Reply:
I think I was going to do a filename selector for the next story? I can split out the header and the content. But it makes sense its in the same section, as it is testing the contents of a single csv file (it's just bloated as the csvheader content is quite large